### DISCUSSION TITLE
Show sprite flags in their byte form

### DISCUSSION THREAD

**Discussion by Weeppiko** [nesbox/TIC-80#2523] [Ideas]
I found myself consistently needing to use flags as index values rather than in it's bitfield form, so it would help a lot if it had a field for typing the byte value in the editor instead of having to manually place each bit for a specific value. Something like this:
![image](https://github.com/nesbox/TIC-80/assets/88540876/e7ada354-2cf0-44e2-ac1b-257699c5da2d)

**Comment by joshgoebel** [comment]
Explain further?  Each sprite already has an index.

**Reply by Weeppiko** [reply]
The suggestion is only about adding that extra field down there in the editor to optionally edit the *Sprite Flags* in hex form. This has nothing to do with Sprite ID's. "Using them as index" is just my use case for this.

**Reply by joshgoebel** [reply]
> Using them as index" is just my use case for this.

I think my point was it's a very uncommon use case... that there isn't really a reason to show the flags in hex form.

**Reply by Weeppiko** [reply]
Not that uncommon for me personally: The collision system I use is not a simple yes/no check, the sprite flag byte is used to index a table containing just a few height data for slopes that are reused over multiple tiles. Currently I'm simply setting the bits to the corresponding index manually, and reading it from RAM using `peek`. This was the most practical way to do it, as I don't need to ever touch the slope table or any other table for new or modified tiles, and it simply works nicely between multiple banks that have different tileset layouts, because it doesn't rely on the Sprite ID's.

**Reply by Weeppiko** [reply]
Even outside of my use case, using the sprite flags as a behavior ID rather than "this is the solid flag", "this is the coin flag", "this is the breakable flag" can be something desirable, since it can simply ignore combinations like *solid coin* or *breakable but non solid block*. And taking an example for something like Mario, a specific hex value can used to mean a question block with a different power up.

**Reply by joshgoebel** [reply]
A table of static flags (in your program) [ie 59 has a specific meaning vs 31] vs bit based flags is persuasive I think. Nesbox would probably be open to a PR for this. 

Of course using a nibble for a table of 16 items plus 4 global flags is a nice compromise also.

**Comment by nesbox** [comment]
done here https://github.com/nesbox/TIC-80/pull/2527
![video1](https://github.com/nesbox/TIC-80/assets/1101448/806fffe2-7c4a-42e4-978b-5df11ccfec94)

**Reply by Weeppiko** [reply]
Thanks, that is going to be really helpful to my workflow with them

**Comment by nesbox** [comment]
Also added the hex index of the current sprite
![video2](https://github.com/nesbox/TIC-80/assets/1101448/a6c26472-ef93-4608-9073-778f4b14b9d5)

**Comment by joshgoebel** [comment]
Awesome work @nesbox
